apps/src/p5lab/AnimationPicker/AnimationPickerBody.jsx

```diff
@@ -101,89 +173,72 @@ class AnimationPickerBody extends React.Component {
           <div style={animationPickerStyles.navigation}>
-            {this.state.categoryQuery !== '' && (
+            {categoryQuery !== '' && (
               <div style={animationPickerStyles.breadCrumbs}>
                 <span
                   onClick={this.onClearCategories}
                   style={animationPickerStyles.allAnimations}
                 >
                   {'All categories > '}
                 </span>
-                <span>{categories[this.state.categoryQuery]}</span>
-              </div>
-            )}
-            {(this.state.searchQuery !== '' ||
-              this.state.categoryQuery !== '') && (
-              <div style={animationPickerStyles.pagination}>
-                <PaginationWrapper
-                  totalPages={pageCount}
-                  currentPage={this.state.currentPage + 1}
-                  onChangePage={this.onChangePageNumber}
-                />
+                <span>{categories[categoryQuery]}</span>
               </div>
             )}
           </div>
         )}
-        <ScrollableList style={{maxHeight: 420}}>
+        <ScrollableList
+          style={{maxHeight: MAX_HEIGHT}}
+          onScroll={this.handleScroll}
+        >
           {' '}
-          {/* TODO: Is this maxHeight appropriate? */}
-          {pageCount === 0 && (
-            <div style={animationPickerStyles.emptyResults}>
-              Sorry, no results found.
-            </div>
-          )}
-          {((this.state.searchQuery === '' &&
-            this.state.categoryQuery === '') ||
-            pageCount === 0) && (
+          {(searchQuery !== '' || categoryQuery !== '') &&
+            results.length === 0 && (
+              <div style={animationPickerStyles.emptyResults}>
+                Sorry, no results found.
+              </div>
+            )}
+          {((searchQuery === '' && categoryQuery === '') ||
+            results.length === 0) && (
             <div>
               <AnimationPickerListItem
                 label={msg.animationPicker_drawYourOwn()}
                 icon="pencil"
-                onClick={this.props.onDrawYourOwnClick}
+                onClick={onDrawYourOwnClick}
               />
               {!hideUploadOption && (
                 <AnimationPickerListItem
                   label={msg.animationPicker_uploadImage()}
                   icon="upload"
-                  onClick={this.props.onUploadClick}
+                  onClick={onUploadClick}
                 />
               )}
             </div>
           )}
-          {this.state.searchQuery === '' &&
-            this.state.categoryQuery === '' &&
+          {searchQuery === '' &&
+            categoryQuery === '' &&
             this.animationCategoriesRendering()}
-          {(this.state.searchQuery !== '' || this.state.categoryQuery !== '') &&
-            this.animationItemsRendering(results)}
+          {(searchQuery !== '' || categoryQuery !== '') &&
```

Comment:
NIT: (up to you) It looks like we're using this `(searchQuery !== '' || categoryQuery !== '')` in several places. It would be nice to factor it into a function.

Reply:
similar comment for `(searchQuery === '' && categoryQuery === '')` - again, up to you.